File: app/src/main/java/com/skydoves/waterdays/ui/viewholders/AlarmViewHolder.java
package com.skydoves.waterdays.ui.viewholders;

import android.content.DialogInterface;
import android.support.annotation.NonNull;
import android.support.v7.app.AlertDialog;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

import com.skydoves.waterdays.R;
import com.skydoves.waterdays.models.AlarmModel;

import butterknife.BindView;

/**
 * Developed by skydoves on 2017-08-20.
 * Copyright (c) 2017 skydoves rights reserved.
 */

public class AlarmViewHolder extends BaseViewHolder {

    protected @BindView(R.id.item_alarmrecord_tv_days) TextView tv_item_days;
    protected @BindView(R.id.item_alarmrecord_tv_times) TextView tv_item_times;
    protected @BindView(R.id.item_alarmrecord_tv_interval) TextView tv_item_interval;
    protected @BindView(R.id.item_alarmrecord_btn_delete) ImageButton ibtn_item_delete;

    private AlarmModel alarmModel;

    private Delegate delegate;

    public interface Delegate {
        void onConfirm(AlarmModel alarmModel);
    }

    public AlarmViewHolder(@NonNull View view, Delegate delegate) {
        super(view);
        this.delegate = delegate;
    }

    @Override
    public void bindData(@NonNull Object data) throws Exception {
        this.alarmModel = (AlarmModel) data;

        tv_item_days.setText(alarmModel.getDays());
        tv_item_times.setText(alarmModel.getTimes());
        tv_item_interval.setText(alarmModel.getInterval());
        ibtn_item_delete.setOnClickListener(view -> {
            AlertDialog.Builder alertDlg = new AlertDialog.Builder(view.getContext());
            alertDlg.setTitle("알람");
            alertDlg.setPositiveButton("예", (DialogInterface dialog, int which) -> delegate.onConfirm(alarmModel));
            alertDlg.setNegativeButton("아니오", ((dialog, which) -> dialog.dismiss()));
            alertDlg.setMessage(String.format("해당 알람을 지우시겠습니까?"));
            alertDlg.show();
        });
    }

    @Override
    public void onClick(View v) {
    }
}

File: app/src/main/java/com/skydoves/waterdays/ui/viewholders/AlarmViewHolder.kt
/*
 * Copyright (C) 2016 skydoves
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.skydoves.waterdays.ui.viewholders

import android.content.DialogInterface
import android.view.View
import androidx.appcompat.app.AlertDialog
import com.skydoves.waterdays.models.Alarm
import kotlinx.android.synthetic.main.item_alarmrecord.view.*

/**
 * Developed by skydoves on 2017-08-20.
 * Copyright (c) 2017 skydoves rights reserved.
 */

class AlarmViewHolder(view: View, private val delegate: Delegate) : BaseViewHolder(view) {

  private lateinit var alarmModel: Alarm

  interface Delegate {
    fun onConfirm(alarmModel: Alarm)
  }

  @Throws(Exception::class)
  override fun bindData(data: Any) {
    this.alarmModel = data as Alarm

    itemView.item_alarmrecord_tv_days.text = alarmModel.days
    itemView.item_alarmrecord_tv_times.text = alarmModel.times
    itemView.item_alarmrecord_tv_interval.text = alarmModel.interval
    itemView.item_alarmrecord_btn_delete.setOnClickListener { view ->
      val alertDlg = AlertDialog.Builder(view.context)
      alertDlg.setTitle("알람")
      alertDlg.setPositiveButton("예") { dialog: DialogInterface, which: Int -> delegate.onConfirm(alarmModel) }
      alertDlg.setNegativeButton("아니오") { dialog, which -> dialog.dismiss() }
      alertDlg.setMessage(String.format("해당 알람을 지우시겠습니까?"))
      alertDlg.show()
    }
  }

  override fun onClick(v: View) {}

  override fun onLongClick(v: View): Boolean {
    return false
  }
}
